src/test/java/stepDefinition/GenericsStepDef.java

```diff
@@ -26,11 +26,17 @@ public void pageIsDisplayed(String pageName) {
 
     @When("^user press on '(.*)' (Button|Field|Checkbox)$")
     public void userPressOnButton(String elementName, String extension) {
-        implicitlyWait(5);
-        waitUntilElementIsClickable(getElementByName(elementName.concat(extension)));
+        waitVisibility(getElementByName(elementName.concat(extension)));
         clickOnElement(getElementByName(elementName.concat(extension)));
     }
 
+    @When("^user clicks on '(.*)' category from category bar$")
+    public void clickOnCategory(String categoryName) {
```

Comment:
you added generic method for navigation, aren't you?

Reply:
Done